crates/xds/src/server.rs

```diff
@@ -180,43 +172,42 @@ impl<C: crate::config::Configuration> ControlPlane<C> {
                 client: ClientVersions::new(ResourceType::Cluster),
                 subscribed: BTreeSet::new(),
                 kind: ResourceType::Cluster,
+                subbed: false,
             },
             ResourceType::Listener => {
                 ResourceTypeTracker {
                     client: ClientVersions::new(ResourceType::Listener),
                     subscribed: BTreeSet::new(),
                     kind: ResourceType::Listener,
+                    subbed: false,
                 }
             },
             ResourceType::FilterChain => {
                 ResourceTypeTracker {
                     client: ClientVersions::new(ResourceType::FilterChain),
                     subscribed: BTreeSet::new(),
                     kind: ResourceType::FilterChain,
+                    subbed: false,
                 }
             },
             ResourceType::Datacenter => {
                 ResourceTypeTracker {
                     client: ClientVersions::new(ResourceType::Datacenter),
                     subscribed: BTreeSet::new(),
                     kind: ResourceType::Datacenter,
+                    subbed: false,
                 }
             }
         };
 
-        let mut cluster_rx = self.watchers[ResourceType::Cluster].receiver.clone();
-        let mut listener_rx = self.watchers[ResourceType::Listener].receiver.clone();
-        let mut fc_rx = self.watchers[ResourceType::FilterChain].receiver.clone();
-        let mut dc_rx = self.watchers[ResourceType::Datacenter].receiver.clone();
-
```

Comment:
Changes these from separate watchers to a single broadcast channel, much more easy to deal with.